File: process_product.py
import json

def process_product_data(input_file):
    # Read the input file
    with open(input_file, 'r') as file:
        lines = file.readlines()
    
    # Get headers from first line
    headers = lines[0].strip().split(',')
    
    # Get data from second line
    data = lines[1].strip().split(',')
    
    # Create dictionary by combining headers and data
    product_dict = {}
    for i, header in enumerate(headers):
        # Handle special cases where data might contain commas within quotes
        if i < len(data):
            value = data[i]
            # Remove quotes if present
            if value.startswith('"') and value.endswith('"'):
                value = value[1:-1]
            product_dict[header] = value
    
    # Convert to JSON
    json_output = json.dumps(product_dict, indent=2)
    
    # Write to output file
    with open('product.json', 'w') as file:
        file.write(json_output)
    
    return json_output

if __name__ == "__main__":
    result = process_product_data('product.txt')
    print("JSON conversion completed. Output saved to product.json") 

File: process_product.js
const fs = require('fs');

function processProductData(inputFile) {
    try {
        // Read the input file
        const content = fs.readFileSync(inputFile, 'utf8');
        const lines = content.split('\n');

        // Get headers from first line
        const headers = lines[0].trim().split(',');

        // Get data from second line
        const data = lines[1].trim().split(',');

        // Create object by combining headers and data
        const productObj = {};
        headers.forEach((header, index) => {
            if (index < data.length) {
                let value = data[index];
                // Remove quotes if present
                if (value.startsWith('"') && value.endsWith('"')) {
                    value = value.slice(1, -1);
                }
                productObj[header] = value;
            }
        });

        // Convert to JSON
        const jsonOutput = JSON.stringify(productObj, null, 2);

        // Write to output file
        fs.writeFileSync('product.json', jsonOutput);

        console.log('JSON conversion completed. Output saved to product.json');
        return jsonOutput;
    } catch (error) {
        console.error('Error processing file:', error.message);
    }
}

// Run the function
processProductData('product.txt'); 